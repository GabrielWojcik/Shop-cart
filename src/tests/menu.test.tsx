import { NavLink } from "react-router-dom"
import MenuMain from "../components/navbar/index"
import { fireEvent, render, screen } from "@testing-library/react"
import nav from "../hooks/router"

const mockNavigation = jest.fn()
const mockNavLink = jest.fn()
const mockRouterNav = jest.fn()

jest.mock("react-router-dom", () => {
    return {
        useNavigate: () => mockNavigation
    }
})

jest.mock('react-router-dom', () => {
    return {
        NavLink: () => mockNavLink
    }
})

jest.mock('../hooks/router', () => {
    return {
        useRouterNavBar: () => mockRouterNav
    }
})

describe('', () => {
    it('Test router dom in navbar component', () => {
        render(<MenuMain /> )
        fireEvent.click(screen.getByText(/sacola/i))
        expect(mockRouterNav).toHaveBeenCalledTimes(1)
        expect(mockNavigation).toHaveBeenCalledTimes(1)
    })
})