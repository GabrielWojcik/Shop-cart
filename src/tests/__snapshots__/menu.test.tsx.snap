// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Test router dom in navbar component 1`] = `null`;
